tests/instancetype_test.go

```diff
@@ -1159,10 +1430,364 @@ var _ = Describe("[crit:medium][vendor:cnv-qe@redhat.com][level:component][sig-c
+			_, err := virtClient.GeneratedKubeVirtClient().InstancetypeV1alpha1().VirtualMachinePreferences(util.NamespaceTestDefault).Get(context.Background(), objName, metav1.GetOptions{})
+			Expect(err).ToNot(HaveOccurred())
+		}
+
+		fetchVirtualMachinePreferencev1alpha2 := func(objName string) {
+			_, err := virtClient.GeneratedKubeVirtClient().InstancetypeV1alpha2().VirtualMachinePreferences(util.NamespaceTestDefault).Get(context.Background(), objName, metav1.GetOptions{})
+			Expect(err).ToNot(HaveOccurred())
+		}
+
+		fetchVirtualMachinePreferencev1beta1 := func(objName string) {
+			_, err := virtClient.VirtualMachinePreference(util.NamespaceTestDefault).Get(context.Background(), objName, metav1.GetOptions{})
+			Expect(err).ToNot(HaveOccurred())
+		}
+
+		fetchVirtualMachineClusterPreferencev1alpha1 := func(objName string) {
+			_, err := virtClient.GeneratedKubeVirtClient().InstancetypeV1alpha1().VirtualMachineClusterPreferences().Get(context.Background(), objName, metav1.GetOptions{})
+			Expect(err).ToNot(HaveOccurred())
+		}
+
+		fetchVirtualMachineClusterPreferencev1alpha2 := func(objName string) {
+			_, err := virtClient.GeneratedKubeVirtClient().InstancetypeV1alpha2().VirtualMachineClusterPreferences().Get(context.Background(), objName, metav1.GetOptions{})
+			Expect(err).ToNot(HaveOccurred())
+		}
+
+		fetchVirtualMachineClusterPreferencev1beta1 := func(objName string) {
+			_, err := virtClient.VirtualMachineClusterPreference().Get(context.Background(), objName, metav1.GetOptions{})
+			Expect(err).ToNot(HaveOccurred())
+		}
+
+		DescribeTable("should create", func(createFunc func() string, v1alpha1FetchFunc func(string), v1alpha2FetchFunc func(string), v1beta1FetchFunc func(string)) {
+			// Create the object and then fetch it using the currently supported versions
+			objName := createFunc()
+			v1alpha1FetchFunc(objName)
+			v1alpha2FetchFunc(objName)
+			v1beta1FetchFunc(objName)
+		},
+			Entry("VirtualMachineInstancetype v1alpha1 and fetch using v1alpha1, v1alpha2 and v1beta1",
+				func() string {
+					createdObj, err := virtClient.GeneratedKubeVirtClient().InstancetypeV1alpha1().VirtualMachineInstancetypes(util.NamespaceTestDefault).Create(context.Background(), &instancetypev1alpha1.VirtualMachineInstancetype{
+						TypeMeta: metav1.TypeMeta{
+							APIVersion: instancetypev1alpha1.SchemeGroupVersion.String(),
+							Kind:       "VirtualMachineInstancetype",
+						},
+						ObjectMeta: metav1.ObjectMeta{
+							GenerateName: "instancetype",
+						},
+						Spec: instancetypev1alpha1.VirtualMachineInstancetypeSpec{
+							CPU: instancetypev1alpha1.CPUInstancetype{
+								Guest: 1,
+							},
+							Memory: instancetypev1alpha1.MemoryInstancetype{
+								Guest: resource.MustParse("128Mi"),
+							},
+						},
+					}, metav1.CreateOptions{})
+					Expect(err).ToNot(HaveOccurred())
+					return createdObj.Name
+				},
+				fetchVirtualMachineInstancetypev1alpha1,
```

Comment:
Should the fetch/verify funcs also assert on the values like `CPU` or `Memory` which were set above?

Reply:
No I don't think so, that's already covered elsewhere and would complicate these tests IMHO.